### PR TITLE
Rearchitect site for npm publishing

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

After digging into a little bit, I realised that the structure scaffolded out from `create-react-app` is not optimal for publishing a component library to NPM.

[1] https://github.com/facebookincubator/create-react-app/issues/907#issuecomment-253998607

On the same thread, I noted where a team had forked create-react-app however for exactly this purpose so I've pulled out the work I've completed up until now and dropped it into that structure instead with Percy and Storybook.

Everything looks in order and the lib now builds as I'd expect and I can publish to npm, however, the app does not seem to be exporting modules correctly as when I try to import it and use a Button component, this happens:

https://codesandbox.io/s/03jv971v3n

Any ideas @anthonydillon or @Lukewh ?

Fixes https://github.com/vanilla-framework/vanilla-framework-react/issues/40

### CODE REVIEW

**Comment by WillMoggridge** [comment]
@barrymcgee Your published package doesn't have an `build/index.js`

package.json:
`"main": "build/index.js",`

Checking package:
```
> ls node_modules/vanilla-framework-react/build/index.js                                            
ls: cannot access 'node_modules/vanilla-framework-react/build/index.js': No such file or directory
```

**Comment by WillMoggridge** [comment]
The build created appears to be a compiled version of the site (I'm not sure what the site is in this case).

I don't think we want to upload this to NPM.
In react-bootstrap, the main entrypoint is set to `lib/index.js`. I think we can follow this pattern. The build for publishing to npm should probably just be the readme, license, `package.json` and the `lib` folder.

https://github.com/react-bootstrap/react-bootstrap/blob/master/package.json

This could be changed after this PR to unblock others. In that case I am happy with this.

**Comment by anthonydillon** [comment]
@WillMoggridge is this ready for another review? If so, please update the labels.

File: package.json [hunk 1]
```diff
@@ -1,45 +1,132 @@
...
-    "raf": "^3.4.0",
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0",
-    "react-scripts": "1.0.14",
-    "@storybook/react": "^3.2.13",
-    "@storybook/addon-actions": "^3.2.13",
-    "@storybook/addon-links": "^3.2.13",
-    "@storybook/addon-info": "^3.2.14"
-  },
+  "version": "0.0.5",
```
Comment (anthonydillon) [inline package.json]: Dropping the version from `0.1.0` to `0.0.5`?

File: src/lib/components/Accordion/AccordionItem.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import './Accordion.css';
+//import './Accordion.css';
```
Comment (anthonydillon) [inline src/lib/components/Accordion/AccordionItem.js]: Should this be commented out?

File: package.json (hunk 1 above)
Comment (WillMoggridge) [reply package.json]: I think this is fine, that is the current version on NPM and `0.0.x` may be a better indication of the project.

File: src/lib/components/Accordion/AccordionItem.js (hunk 2 above)
Comment (WillMoggridge) [reply src/lib/components/Accordion/AccordionItem.js]: I think it should not be commented and explicitly load the CSS.

**Pull request merged**